src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,8 +1,26 @@
-export const TodoFilter = () => (
+type Props = {
+  selectedFilter: string,
+  setSelectedFilter: (select: string) => void,
+  query: string,
+  setQuery: (searchText: string) => void,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  selectedFilter,
+  setSelectedFilter,
+  query,
+  setQuery,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
+        <select
+          data-cy="statusSelect"
+          value={selectedFilter}
+          onChange={(event) => {
+            setSelectedFilter(event.target.value);
+          }}
```

Comment:
create handlers instead of anonymous function (+ similar cases)
```suggestion
          onChange={handleSelectChange}
```

Reply:
done